python/cuxfilter/charts/datashader/plots.py

```diff
@@ -534,6 +548,437 @@ def apply_theme(self, properties_dict):
+            callback(xmin, xmax, ymin, ymax)
+
+        self.chart.on_event(events.SelectionGeometry, temp_callback)
+
+    def apply_theme(self, properties_dict):
+        """
+        apply thematic changes to the chart based on the input
+        properties dictionary.
+        """
+        if self.no_colors_set:
+            self.node_color_palette = properties_dict["chart_color"][
+                "color_palette"
+            ]
+            self.interactive_image.update_chart()
+        self.chart.xgrid.grid_line_color = properties_dict["geo_charts_grids"][
+            "xgrid"
+        ]
+        self.chart.ygrid.grid_line_color = properties_dict["geo_charts_grids"][
+            "ygrid"
+        ]
+
+        # title
+        self.chart.title.text_color = properties_dict["title"]["text_color"]
+        self.chart.title.text_font = properties_dict["title"]["text_font"]
+        self.chart.title.text_font_style = properties_dict["title"][
+            "text_font_style"
+        ]
+        self.chart.title.text_font_size = properties_dict["title"][
+            "text_font_size"
+        ]
+
+        # background, border, padding
+        self.chart.background_fill_color = properties_dict[
+            "background_fill_color"
+        ]
+        self.chart.border_fill_color = properties_dict["border_fill_color"]
+        self.chart.min_border = properties_dict["min_border"]
+        self.chart.outline_line_width = properties_dict["outline_line_width"]
+        self.chart.outline_line_alpha = properties_dict["outline_line_alpha"]
+        self.chart.outline_line_color = properties_dict["outline_line_color"]
+
+        # x axis title
+        self.chart.xaxis.major_label_text_color = properties_dict["xaxis"][
+            "major_label_text_color"
+        ]
+        self.chart.xaxis.axis_line_width = properties_dict["xaxis"][
+            "axis_line_width"
+        ]
+        self.chart.xaxis.axis_line_color = properties_dict["xaxis"][
+            "axis_line_color"
+        ]
+
+        # y axis title
+        self.chart.yaxis.major_label_text_color = properties_dict["yaxis"][
+            "major_label_text_color"
+        ]
+        self.chart.yaxis.axis_line_width = properties_dict["yaxis"][
+            "axis_line_width"
+        ]
+        self.chart.yaxis.axis_line_color = properties_dict["yaxis"][
+            "axis_line_color"
+        ]
+
+        # axis ticks
+        self.chart.axis.major_tick_line_color = properties_dict["axis"][
+            "major_tick_line_color"
+        ]
+        self.chart.axis.minor_tick_line_color = properties_dict["axis"][
+            "minor_tick_line_color"
+        ]
+        self.chart.axis.minor_tick_out = properties_dict["axis"][
+            "minor_tick_out"
+        ]
+        self.chart.axis.major_tick_out = properties_dict["axis"][
+            "major_tick_out"
+        ]
+        self.chart.axis.major_tick_in = properties_dict["axis"][
+            "major_tick_in"
```

Comment:
Perhaps open an issue on the Bokeh repo about mass updates of properties. I feel like this this should be made possible with some kind of list of properties to copy passed to a single function call, or maybe a loop over the same. (Just not so much explicit repeated boilerplate)

Reply:
Mass update seems a good idea for bokeh. I'll raise an issue. Thanks!

Reply:
All of the charts have this when applying the theme parameters. I'll follow up with a bokeh issue and a separate PR for this